aten/src/THCUNN/SpatialClassNLLCriterion.cu

```diff
@@ -103,8 +103,6 @@ __global__ void cunn_SpatialClassNLLCriterion_updateOutput_kernel(
     }
   }
 
-  __syncthreads();
-
   input_sum = reduceBlock(partial_sums, blockDim.x, input_sum, thrust::plus<AccumT>(), AccumT(0));
   acc_weight = reduceBlock(partial_sums, blockDim.x, acc_weight, thrust::plus<AccumT>(), AccumT(0));
```

Comment:
We need a __syncthreads() in between these two, because partial_sums is reused, yes?

Reply:
ah good point!